File: 프로그래머스/lv1/42862. 체육복/체육복.py
def solution(n, lost, reserve):   
    answer=n-len(lost)
    
    lost.sort()
    reserve.sort()
    
    for i in lost:
        if i in reserve:
            answer=answer+1
            reserve.remove(i)
            lost.remove(i)
            
    for i in lost:
        if i in reserve:
            answer=answer+1
            reserve.remove(i)
        elif (i-1) in reserve:
            answer=answer+1
            reserve.remove(i-1)
        elif (i+1) in reserve:
            answer=answer+1
            reserve.remove(i+1)
            
    return answer

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int[] people = new int[n+2];
        Arrays.fill(people, 1);
        for(int l:lost){
            people[l]--;
        }
        for(int r:reserve){
            people[r]++;
        }
        
        for(int i= 1; i<=n; i++){
            if(people[i] == 0){
                if(people[i-1]==2){
                    people[i]++;
                    people[i-1]--;
                }
                else if(people[i+1]==2){
                    people[i]++;
                    people[i+1]--;
                }
            }
        }
        int answer = 0;
        for(int i=1; i<=n; i++){
            if(people[i]>=1) answer++;
        }
        return answer;
    }
}